### ISSUE TITLE
"set cmd:fail-exit yes" causes random exits on sftp connection

### ISSUE THREAD

**Issue by L-B** [lavv17/lftp#128]
When using "set cmd:fail-exit yes" on an sftp connection, lftp exits to the shell randomly after a few commands. It happens apparently on any command, even some that can't possibly fail, like "help".
There is no error message, nor anything obviously wrong in the logs.

I found this on RHEL6's RPM, which is at 4.0.9, then built the latest 4.6.1, and got the same issue.
Please note that a case was raised with RH,  BZ#1193617 (private).

Here is an example of a session, and how it ends:
```
login@hostname:~ $ ./lftp-4.6.1/bin/lftp -e "set cmd:fail-exit yes; debug -o /var/tmp/log_$(date +%Y%m%d-%H%M%S).debug;" sftp://hostname
lftp hostname:~> ls
<list of files>
lftp hostname:~> cls
<list of files>
lftp hostname:~> help ls
Usage: ls [<args>]
List remote files. You can redirect output of this command to file
or via pipe to external command.
By default, ls output is cached, to see new listing use `rels' or
`cache flush'.
See also `help cls'.
lftp hostname:~> help cls
login@hostname:~ $ echo $?
1
```

And this is the content of the debug file:
```
login@hostname:/var/tmp $ cat log_20150218-100220.debug
---- Running connect program (ssh -a -x -s hostname sftp)
---> sending a packet, length=5, type=1(INIT), id=0
<--- 
<--- got a packet, length=95, type=2(VERSION), id=0
---- protocol version set to 3
---> sending a packet, length=10, type=16(REALPATH), id=1
<--- got a packet, length=45, type=104(NAME), id=1
---- home set to /home/login
---- path on wire is `/home/login'
---> sending a packet, length=21, type=11(OPENDIR), id=2
<--- got a packet, length=13, type=102(HANDLE), id=2
---- got file handle 00000000 (4)
---> sending a packet, length=13, type=12(READDIR), id=3
<--- got a packet, length=6967, type=104(NAME), id=3
---- file name count=58
---> sending a packet, length=13, type=12(READDIR), id=4
---> sending a packet, length=13, type=12(READDIR), id=5
<--- got a packet, length=28, type=101(STATUS), id=4
---- status code=1(EOF), message=End of file
---- eof
<--- got a packet, length=28, type=101(STATUS), id=5
---- status code=1(EOF), message=End of file
---> sending a packet, length=13, type=4(CLOSE), id=6
<--- got a packet, length=24, type=101(STATUS), id=6
---- status code=0(OK), message=Success
---- Disconnecting
```

**Comment by lavv17** [comment]
"help" exit code fixed in 03d6b34. For sftp it's not clear which commands are being executed, please send the complete script.

Note that when "fail-exit" is true, lftp exits not after failed command, but before the next one. So lftp does not exit until you type a next command (not prefixed with ||).

**Comment by L-B** [comment]
Oh, that was the complete script.
But what you're saying makes it clear: it's not random, I was just not looking at the right line.
The trigger is "help <something>", that causes the next command to exit back to the shell, whatever it is.
"help" alone does not cause it.
So I gather for this particular issue, that fix is indeed what is needed.
Sadly, it means I will have to dig deeper for some other problems.
Thanks for your help!
